server_plugin/server.py

```diff
@@ -140,24 +140,25 @@ def _create(vca_client, config, server):
     task = vca_client.create_vapp(config['vdc'],
                                   vapp_name,
                                   vapp_template,
-                                  vapp_catalog,
-                                  vm_name=vapp_name)
+                                  vapp_catalog)
     if not task:
         raise cfy_exc.NonRecoverableError("Could not create vApp: {0}"
                                           .format(error_response(vca_client)))
     wait_for_task(vca_client, task)
 
+    vdc = vca_client.get_vdc(config['vdc'])
+    vapp = vca_client.get_vapp(vdc, vapp_name)
+    if vapp is None:
+        raise cfy_exc.NonRecoverableError(
+            "vApp '{0}' could not be found".format(vapp_name))
+
+    vapp.modify_vm_name(1, vapp_name)
```

Comment:
This function returns TaskType.
https://github.com/cloudify-cosmo/tosca-vcloud-plugin/blob/custom/server_plugin/server.py#L150